test/e2e/overrides_test.go

```diff
@@ -0,0 +1,152 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Overrides test", Ordered, func() {
+
+	BeforeEach(func() {
+		By("Removing any existing overrides")
+		err := removeOverrides(certmanageroperatorclient)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for operator status to become available")
+		err = waitForValidOperatorStatusCondition(certmanageroperatorclient,
+			[]string{certManagerControllerDeploymentControllerName,
+				certManagerWebhookDeploymentControllerName,
+				certManagerCAInjectorDeploymentControllerName})
+		Expect(err).NotTo(HaveOccurred(), "Operator is expected to be available")
+	})
+
+	Context("When adding valid cert-manager controller override args", func() {
+
+		It("should add the args to the cert-manager controller deployment", func() {
+
+			By("Adding cert-manager controller override args to the cert-managaer operator object")
+			args := []string{"--dns01-recursive-nameservers=10.10.10.10:53", "--dns01-recursive-nameservers-only", "--enable-certificate-owner-ref", "--v=3"}
+			err := addOverrideArgs(certmanageroperatorclient, certmanagerControllerDeployment, args)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for cert-manager controller status to become available")
+			err = waitForValidOperatorStatusCondition(certmanageroperatorclient, []string{certManagerControllerDeploymentControllerName})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for the args to be added to the cert-manager controller deployment")
+			err = waitForDeploymentArgs(k8sClientSet, certmanagerControllerDeployment, args, true)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
We need to verify the behaviour that was added by the override args not just that the operator is working. We are not performing any verification for the parameters that we are testing. At minimum we need to ensure the values passed as args are present on the operand.

Reply:
The operand deployment is being checked that if the args are added to them or not. Do we need to check the pods as well? IMO as we are only updating the corresponding deployments, it should suffice checking the deployment args only. Additionally, the deployment controller should ensure to trickle down the args to the pods as well. WDYT?

Reply:
Ahh missed it because of the naming convention used. Can we rename all verification functions to have some sort of prefix "verify". Wait doesn't really convey the right intention here.

Reply:
`waitForValidOperatorStatusConditio` for status makes sense as there we are waiting for operator stability. Also can you merge both `waitForInvalidOperatorStatusCondition` and `waitForValidOperatorStatusConditio` to just take a list a status conditions. We can then maintain a list of valid conditions/status?

Reply:
> `waitForValidOperatorStatusConditio` for status makes sense as there we are waiting for operator stability. Also can you merge both `waitForInvalidOperatorStatusCondition` and `waitForValidOperatorStatusConditio` to just take a list a status conditions. We can then maintain a list of valid conditions/status?

Makes sense. Will make this change.

Reply:
Made the changes